ftppush.sh: If one file fails to upload, the file and all others following are marked "already sent"

Hi,

I've got the problem that my ftp server's space is limited and usually a bash script on the FTP server further processes the files sent by the YI camera. If for some reason, the processing stops unexpectedly on the FTP server side, the file systems fills up to 100% if the camera keeps sending mp4 files to it.

I then get a message in "/tmp/ftppush.log" (camera) stating "ignore --- filename --- already sent." and the script does not bump the last-file-sent-timestamp variable. So far so good. Then my FTP server catches up and it gets to 98% free disk space there.

The camera sends another file which is at the moment too big and gets a "0 byte" file on the FTP server. The camera notes success (ftpput seems to return SUCCESS) and the sent-file-timestamp is updated with the "before FAILED" file skipped which is not correct. I've changed single line to prevent this desaster scenario of filling up the SDcard and never getting files sent out. It also makes no sense, if the FTP server encounters an error to keep sending all stuff after the current file. The retry should kick in immediately and give the FTP server (or even network) time to settle things and then retry after 45s.

Will make a PR to improve this.

Kind regards,
Catfriend1


Log after PR:
```
2021-07-25 [20-28-23] [INFO] checkFiles: uploadToFtp SUCCEEDED - [/tmp/sd/record/2021Y07M24D11H/41M00S60.mp4].
257 "/2021Y07M24D12H" created
2021-07-25 [20-28-24] [INFO] checkFiles: uploadToFtp SUCCEEDED - [/tmp/sd/record/2021Y07M24D12H/02M46S14.mp4].
2021-07-25 [20-28-28] [INFO] checkFiles: uploadToFtp SUCCEEDED - [/tmp/sd/record/2021Y07M24D12H/03M00S60.mp4].
2021-07-25 [20-28-30] [INFO] checkFiles: uploadToFtp SUCCEEDED - [/tmp/sd/record/2021Y07M24D12H/05M28S32.mp4].
2021-07-25 [20-28-32] [ERROR] uploadToFtp: ftpput FAILED.
2021-07-25 [20-28-32] [ERROR] checkFiles: uploadToFtp FAILED - [/tmp/sd/record/2021Y07M24D12H/06M00S60.mp4]. Retrying in 45s.
2021-07-25 [20-29-18] [INFO] checkFiles
2021-07-25 [20-29-22] [INFO] checkFiles: uploadToFtp SUCCEEDED - [/tmp/sd/record/2021Y07M24D12H/06M00S60.mp4].
2021-07-25 [20-29-26] [INFO] checkFiles: uploadToFtp SUCCEEDED - [/tmp/sd/record/2021Y07M24D12H/07M10S50.mp4].
2021-07-25 [20-29-30] [INFO] checkFiles: uploadToFtp SUCCEEDED - [/tmp/sd/record/2021Y07M24D12H/08M00S60.mp4].
2021-07-25 [20-29-32] [INFO] checkFiles: uploadToFtp SUCCEEDED - [/tmp/sd/record/2021Y07M24D12H/15M40S20.mp4].
2021-07-25 [20-29-36] [INFO] checkFiles: uploadToFtp SUCCEEDED - [/tmp/sd/record/2021Y07M24D12H/16M00S60.mp4].
2021-07-25 [20-29-36] [ERROR] uploadToFtp: ftpput FAILED.
2021-07-25 [20-29-36] [ERROR] checkFiles: uploadToFtp FAILED - [/tmp/sd/record/2021Y07M24D12H/19M18S42.mp4]. Retrying in 45s.
2021-07-25 [20-30-22] [INFO] checkFiles
2021-07-25 [20-30-25] [INFO] checkFiles: uploadToFtp SUCCEEDED - [/tmp/sd/record/2021Y07M24D12H/19M18S42.mp4].
2021-07-25 [20-30-30] [INFO] checkFiles: uploadToFtp SUCCEEDED - [/tmp/sd/record/2021Y07M24D12H/20M00S60.mp4].
2021-07-25 [20-30-34] [INFO] checkFiles: uploadToFtp SUCCEEDED - [/tmp/sd/record/2021Y07M24D12H/21M00S60.mp4].
2021-07-25 [20-30-37] [INFO] checkFiles: uploadToFtp SUCCEEDED - [/tmp/sd/record/2021Y07M24D12H/28M08S52.mp4].
2021-07-25 [20-30-40] [ERROR] uploadToFtp: ftpput FAILED.
2021-07-25 [20-30-40] [ERROR] checkFiles: uploadToFtp FAILED - [/tmp/sd/record/2021Y07M24D12H/29M00S60.mp4]. Retrying in 45s.
```

--- a/src/static/static/home/yi-hack/script/ftppush.sh
+++ b/src/static/static/home/yi-hack/script/ftppush.sh
@@ -70,8 +70,8 @@ checkFiles ()
 		LAST_FILE_SENT_REMPART=${LAST_FILE_SENT:5:2}${LAST_FILE_SENT:8:2}${LAST_FILE_SENT:11:2}${LAST_FILE_SENT:14:2}
 		if [ ${FILE_YEAR} -gt ${LAST_FILE_SENT_YEAR} ] || ( [ ${FILE_YEAR} -eq ${LAST_FILE_SENT_YEAR} ] && [ ${FILE_REMPART} -gt ${LAST_FILE_SENT_REMPART} ] ); then
 			if ( ! uploadToFtp -- "${file}" ); then
-				logAdd "[ERROR] checkFiles: uploadToFtp FAILED - [${file}]."
-				continue
+				logAdd "[ERROR] checkFiles: uploadToFtp FAILED - [${file}]. Retrying in ${SLEEP_CYCLE_SECONDS} s."
+				return 0
 			fi
 			logAdd "[INFO] checkFiles: uploadToFtp SUCCEEDED - [${file}]."
 			LAST_FILE_SENT=${FILE_DATE}

